packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -138,12 +132,120 @@ export class Relayer {
         );
       }
     }
+
+    return;
+  }
+
+  async checkForPendingRelaysAndDispute(): Promise<undefined> {
+    this.logger.debug({ at: "Disputer", message: "Checking for pending relays and disputing" });
+
+    // Build dictionary of pending relays keyed by l1 token and deposit hash. We assume that getPendingRelays() filters
+    // out Finalized relays.
+    const pendingRelays: Relay[] = this.getPendingRelays();
+    if (pendingRelays.length == 0) {
+      this.logger.debug({ at: "Disputer", message: "No pending relays" });
+      return;
+    }
+    this.logger.debug({
+      at: "Disputer",
+      message: `Processing ${pendingRelays.length} pending relays`,
+    });
+
+    for (const relay of pendingRelays) {
+      // Construct deposit from relay params if the deposit is undefined.
+      const deposit =
```

Comment:
This looks like it looks at _all_ relays, but then only checks against a _single_ l2 client? If the L2 client is pointed at arbitrum, but the bot sees a valid relay from optimism, will it dispute it?

Put another way, how do we deal with multiple L2s here?

Reply:
Really good catch @mrice32 , with the latest commits do you think there are any weird edge cases now where we can misconfigure the L2/L1 clients and create false disputes unintentionally?

@chrismaree worth reviewing this thread and re reviewing